tf2_tools/tf2_tools/view_frames.py

```diff
@@ -85,9 +86,21 @@ def main():
         node.get_logger().info(
             'Result:'+ str(result) )
         data = yaml.safe_load(result.frame_yaml)
-        with open('frames.gv', 'w') as f:
-           f.write(generate_dot(data, node.get_clock().now().seconds_nanoseconds()))
-        subprocess.Popen('dot -Tpdf frames.gv -o frames.pdf'.split(' ')).communicate()
+        
+        if parsed_args.output is not None:
+            datetime = time.strftime('%Y-%m-%d_%H.%M.%S')
+            frames_gv = '{:s}_{:s}.gv'.format(parsed_args.output, datetime)
+            frames_pdf = '{:s}_{:s}.pdf'.format(parsed_args.output, datetime)
```

Comment:
Almost.  If the user specifies a filename, I think that we should honor it exactly without further changes.  So this should be:

```
frames_gv = '{:s}.gv'.format(parsed_args.output)
framed_pdf = '{:s}.pdf'.format(parsed_args.output)
```

Reply:
Ok, I was trying to be fancy with adding a timedstamp to it, if the user runs the same command multiple times with same filename.

Reply:
Hmm, that's a valid concern, but also the user should shoulder such a burden.

Reply:
Yeah, agreed.  My basic thinking on this is that by default, we should be sure not to overwrite data.  But if the user explicitly tells us to overwrite something, we should just go ahead and do it.